### PR TITLE
repoquery: --unsatisfied and --latest-limit

### PR DESCRIPTION
**Pull request by MichaelMraka**
an implementation of package-cleanup --problems and support for package-cleanup --cleandups

### CODE REVIEW

**Comment by MichaelMraka** [comment]
So I moved goal.run(verify=True) completely to repoquery.

File: doc/repoquery.rst [hunk 1]
```diff
@@ -57,10 +57,18 @@ Together with ``<pkg-spec>``, control what packages are displayed in the output.
 ``--installonly``
     Limit the resulting set to installed installonly packages.
 
+``--latest-limit <number>``
+    Limit the resulting set to <number> of latest packages for every package name and architecture.
+    If <number> is negative skip <number of latest packages.
```
Comment (jsilhan) [inline doc/repoquery.rst]: `<number` -> `<number>`

File: doc/repoquery.rst [hunk 2]
```diff
@@ -133,3 +141,11 @@ Display all available packages providing "webserver"::
...
 
     dnf repoquery --whatprovides webserver --arch i686
+
+Display duplicated packages::
+
+    dnf repoquery --duplicated
+
+Remove older versions of duplicated packages (an equivalent of yum's `package-cleanup --cleandups`)::
+
+    idnf remove $(dnf repoquery --duplicated --latest-limit -1)
```
Comment (jsilhan) [inline doc/repoquery.rst]: `idnf` - Improved DNF (new fork of DNF)? :)

File: plugins/repoquery.py [hunk 3]
```diff
@@ -114,6 +114,9 @@ def parse_arguments(args):
     outform.add_argument('--qf', "--queryformat", dest='queryformat',
                          default=QFORMAT_DEFAULT,
                          help=_('format for displaying found packages'))
+    outform.add_argument("--latest-limit", dest='latest_limit', type=int,
+                         help=_('show N latest packages for a given name.arch'
+                                + '(or latest but N if N is negative)'))
```
Comment (jsilhan) [inline plugins/repoquery.py]: I would add space before `(`. btw `+` sign is not needed here

**Comment by MichaelMraka** [comment]
fixed typos

**Comment by jsilhan** [comment]
Merged, thanks

**Pull request closed without merging**